### ISSUE TITLE
Date format does not appear to be localized 

### ISSUE THREAD

**Issue by danielbitzer** [woocommerce/woocommerce-admin#3666]
**Describe the bug**
When viewing tables in WC Admin dates seem to always be formatted as `MM/DD/YYYY`. 

**To Reproduce**

1. Go to change store address to an Australian address in WooCommerce > Settings > General
2. Go to Analytics > Orders
3. View order **Date** column
4. Dates will be still in US format.

**Expected behavior**
Date format should be localized for the store's region as in WooCommerce's existing admin pages. 

**Screenshots**
![Screenshot](https://d.pr/i/UG8qtX+)
Screenshot: https://d.pr/i/UG8qtX

**Comment by psealock** [comment]
Thanks for the issue @danielbitzer. Indeed, dates on tables are set to the default instead of a site setting.

**Comment by psealock** [comment]
Here is an example of how we've handled internationalization of date formats:

https://github.com/woocommerce/woocommerce-admin/blob/859d050142de002fb41962f04a8c0887cb28baae/packages/components/src/advanced-filters/date-filter.js#L24-L25

Those data format strings can be translated before being interpreted by moment.js. Here is an example of Spanish translations: https://translate.wordpress.com/projects/wpcom/es/default/?filters%5Bterm%5D=momentjs&filters%5Buser_login%5D=&filters%5Bstatus%5D=current&filters%5Bview%5D=&filter=Filter&sort%5Bby%5D=priority&sort%5Bhow%5D=desc
